[Tool] Build tool page

jennifer-lu opened the pull request:
## Notion ticket link
<!-- Please replace with your ticket's URL -->
[[Tool] Build tool page](https://www.notion.so/uwblueprintexecs/Tool-Build-tool-page-259badfdb1734afa93cfca3bba993ecf)


<!-- Give a quick summary of the implementation details, provide design justifications if necessary -->
## Implementation description
* Built tool page
* Added util helper function for `useMediaQuery` with fixed screen size breakpoint
* Used `ReactPlayer` for displaying video
* Split styling for buttons into `AdminButton` and `Button` component themes
* Added tool layout with navbar


<!-- What should the reviewer do to verify your changes? Describe expected results and include screenshots when appropriate -->
## Steps to test
1. From `http://localhost:3000/` click on 'Continue Learning' for any tool (should take you to `http://localhost:3000/tool?id=[id]`)
2. Observe tool page (please compare to [Figma](https://www.figma.com/file/GhDd9C0XsyVDgwL5zzvDvO/HUG-W22?node-id=4355%3A55348)), should look something like this:

<img width="600" alt="Screen Shot 2022-02-24 at 8 36 01 PM" src="https://user-images.githubusercontent.com/49915445/155636956-33fd0c81-1873-4401-b505-524b0238c066.png"> <img width="160" alt="Screen Shot 2022-02-24 at 8 36 24 PM" src="https://user-images.githubusercontent.com/49915445/155636962-d6bcb6d2-2df2-44ba-a7b6-6b3876458e8c.png">

4. Scroll on page to check that navbar and footer are sticky
5. Observe that the video player is present (the video link may or may not be broken)
6. Observe that there is a description below the video
7. Expand and collapse all additional resources tabs and make sure that show the resource links
8. Click on some resource links and make sure they open in a new tab
9. Observe that there are recommended tool cards
10. Click on 'Start Module' and observe redirect to `http://localhost:3000/module?id=[id]` (note the id in the url is unique, and the link does not yet work)
11. Click on 'Take Self-Check' and observe redirect to `http://localhost:3000/selfCheck?id=[id]` (note the id in the url is unique, and the link does not yet work)
12. Close footer and check that it disappears
13. Click on the certificate icon in the header and check that it toggles a banner
14. Click on 'here' in the banner text and observe redirect to `http://localhost:3000/certificate?id=[id]`
15. Check that the 'Your Module Progress' section has a 100% full progress bar
16. Click the restart icon and check that a modal is opened
17. Click on 'No, cancel' or 'Yes, restart' to close the modal (note that restarting does nothing right now)
18. Click on 'Review Module' and observe redirect to `http://localhost:3000/module?id=[id]`(note the id in the url is unique, and the link does not yet work)
19. Click on 'Take Self-Check' and observe redirect to `http://localhost:3000/selfCheck?id=[id]`(note the id in the url is unique, and the link does not yet work)
20. In the code, search for `progress = 100` in the file `pages/tool.tsx`, and change it to `progress = 21`
21. Check that the 'Your Module Progress' section has a 21% full progress bar and the green button says 'Continue Module'
22. Click on the certificate icon in the header and check that it toggles a banner, and the banner no longer has a 'here' link
23. In the code, search for `progress = 21` in the file `pages/tool.tsx`, and change it to `progress = 0`
24. Check that the 'Your Module Progress' section is now titled 'Interested in learning more about ...', there is no progress bar, and the green button says 'Start Module' (note that this section should be relocated below the video and description for mobile layouts)
25. Repeat steps 2-24 with a mobile layout (resize your window horizontally)
26. Also test more than one module, and also a newly created module


<!-- Draw attention to the substantial parts of your PR or anything you'd like a second opinion on
For Frontend changes be sure to specify which routes/pages reviewers should check out!
 -->
## What should reviewers focus on?
* Does it work?
* Does the styling look ok?
* Does the code look ok?


## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

ChinemeremChigbo commented:
Really great work so far! A couple things I noticed were:
- If the screen size is changed to mobile by resizing the window the mediaquery works as intended, but the screen doesn't revert back to desktop mode when the screen is re-enlargened except when clicking the hamburger menu button
- The save button text is now black (different from main which has white text)
![image](https://user-images.githubusercontent.com/24593519/155653578-0700b8f2-f268-42c6-8bea-38f42ed93bf1.png)
- I think the empty ones should probably be hidden (if there are only 2 related resources only show those 2 if there are none, hide the item)
![image](https://user-images.githubusercontent.com/24593519/155653677-918341a2-1e80-402a-85e4-96b5ea48b454.png)
- Maybe add a hover on this to show that it's clickable 
![image](https://user-images.githubusercontent.com/24593519/155653903-8b6a62e1-8513-446a-871c-9a9be7a75062.png)
- Some of the related tools do not have linked modules. This might be out of scope for this PR, but I think that there should be a filter on the "select related tools" dropwdown to only allow published tools (that way they have linked modules and are ready to be shared with end users)
![image](https://user-images.githubusercontent.com/24593519/155654011-33d1ef39-ed6f-495a-9e9c-c03d52f4a22a.png)
- I think the recommended tools carousel behaviour is a little unexpected. Instead the current wraparound behaviour when an end is hit maybe, just have it stop instead if possible
![image](https://user-images.githubusercontent.com/24593519/155654614-23c594a3-9a4e-4ecf-a3da-e195259493d3.png)

Once again, great work!

Het7 reviewed (commented):
Great work on this @jennifer-lu!

jennifer-lu commented:
Thank you for the detailed and thorough review Chinemerem!

Here are my fixes:
- I broke admin button styling by passing the wrongly named `AdminButton`component in the admin theme; fixed by passing `Button: AdminButton`
- The video, additional resources (including individual empty panels), and recommended tools are now hidden if they are empty
- The certificate icon and tool cards get a lime outline on hover (an outline on hover for tool cards was specified in the Figma but I missed it)
- Removed wrap around behaviour for the carousel (also had to remove `emulateTouch` because it was glitchy, so non touchscreen users will have to click instead of swipe)

I couldn't seem to replicate the responsiveness issue, though I was having some issues with responsiveness in the navbar earlier (which were fixed thanks to Tony's help).

> Some of the related tools do not have linked modules. This might be out of scope for this PR, but I think that there should be a filter on the "select related tools" dropwdown to only allow published tools (that way they have linked modules and are ready to be shared with end users)

This sounds like a great idea, and I can definitely implement it. Just wondering if this needs to be discussed further before I add it in or can I just add it?

jennifer-lu commented:
Thanks @Het7 !

ChinemeremChigbo reviewed (approved):
Looks great! I honestly think you should implement it if you have time, since I don't think an admin would want users to have access to uncomplete modules @Chamod what do you think?

jennifer-lu commented:
> Some of the related tools do not have linked modules. This might be out of scope for this PR, but I think that there should be a filter on the "select related tools" dropwdown to only allow published tools (that way they have linked modules and are ready to be shared with end users)

Discussed this with Chamod, and the chosen solution is to only display published related tools on the page. This way, admins can still link unpublished tools but the end user won't see them until they are published.

tawnyzhao commented on pages/tool.tsx:
```diff
@@ -1,49 +1,239 @@
+import { ToolLayout } from "@components";
+import Footer from "src/pages/tool/Footer";
+import Header from "src/pages/tool/Header";
+import ModuleProgressSection from "src/pages/tool/ModuleProgressSection";
```
Let's use the aliases here for imports. i.e. "src/pages/..." => "@pages/..."

tawnyzhao commented on pages/tool.tsx:
```diff
@@ -1,49 +1,239 @@
+    );
 
-            <br />
+    if (error || relatedToolsError) {
```
If an error occurs with the related tools, we should still render as much as we can. Consider moving the error handling for related tools to be just under the related tools section

tawnyzhao replied:
move to conditional rendering on line 113
